rust/index/src/spann/types.rs

```diff
@@ -742,6 +847,42 @@ impl SpannIndexWriter {
                 next_version,
                 prev_head_id
             );
+            self.metrics.num_reassigns.add(
+                1,
+                &[KeyValue::new(
+                    "collection_id",
+                    self.collection_id.to_string(),
+                )],
+            );
+            match reason {
```

Comment:
Are these "subevents" of num_reassign? You could just store the reason as an attribute and slice/dice on that instead. Suggest that as its less metrics.